### PR TITLE
Fix Key Vault's ShareLink sample

### PR DESCRIPTION
**Pull request by heaths**
This should not only fix the ShareLink sample, but make it more robust against future refactorings. It'll have people add package refs for Azure.Core as well as Microsoft.Azure.AutoRest.CSharp and does add the nuget.config to the ZIP.

### CODE REVIEW

File: sdk/keyvault/samples/sharelink/ShareLink.csproj [hunk 1]
```diff
@@ -30,29 +30,28 @@
...
+    <None Include="..\..\..\..\NuGet.Config" Link="NuGet.Config" />
   </ItemGroup>
 
   <Target Name="CopySource">
     <Error Text="The 'Destination' property must be specified" Condition="'$(Destination)' == ''" />
 
     <ItemGroup>
       <!-- Collect all sources and ensure they have a relative Link metadata property -->
-      <_Source Include="@(Compile->HasMetadata('Link'))" />
+      <_Source Include="@(Compile->HasMetadata('Link'));@(None->HasMetadata('Link'))" />
```
Comment (christothes) [inline sdk/keyvault/samples/sharelink/ShareLink.csproj:54]: what does this to?

**Review by christothes — APPROVED** [review]

File: sdk/keyvault/samples/sharelink/ShareLink.csproj (hunk 1 above)
Comment (heaths) [reply sdk/keyvault/samples/sharelink/ShareLink.csproj:54]: Copies the `ItemGroup` defined above when you copy to a separate directory outside our repo. Mostly, it's a way to test the frontmatter definitions in the README.md for samples.

File: sdk/keyvault/samples/sharelink/ShareLink.csproj [hunk 2]
```diff
@@ -30,29 +30,28 @@
...
-  <ItemGroup>
-    <Compile Include="..\..\..\core\Azure.Core\src\Shared\AuthorizationChallengeParser.cs" LinkBase="Shared\Core" />
-  </ItemGroup>
-
   <ItemGroup Condition="'$(IsSample)' == 'true'">
     <!-- Link files needed for Key Vault -->
+    <Compile Include="..\..\..\core\Azure.Core\src\Shared\AuthorizationChallengeParser.cs" LinkBase="Shared\Core" />
     <Compile Include="..\..\Azure.Security.KeyVault.Shared\src\ChallengeBasedAuthenticationPolicy.cs" LinkBase="Shared\KeyVault" />
+
+    <None Include="..\..\..\..\NuGet.Config" Link="NuGet.Config" />
```
Comment (JoshLove-msft) [inline sdk/keyvault/samples/sharelink/ShareLink.csproj:46]: Why?

**Review by JoshLove-msft — APPROVED** [review]

File: sdk/keyvault/samples/sharelink/ShareLink.csproj (hunk 2 above)
Comment (heaths) [reply sdk/keyvault/samples/sharelink/ShareLink.csproj:46]: Microsoft.Azure.AutoRest.CSharp isn't published to nuget.org.

**Pull request merged**